### PR TITLE
dmd.imphint: Remove C++ linkage from importHint

### PR DESCRIPTION
**Pull request by ibuclaw**
Update signature to return a const(char)[] instead.

### CODE REVIEW

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -2326,8 +2326,8 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
         /* Look for what user might have meant
          */
-        if (const n = importHint(exp.ident.toChars()))
-            exp.error("`%s` is not defined, perhaps `import %s;` is needed?", exp.ident.toChars(), n);
+        if (const n = importHint(exp.ident.toString()))
+            exp.error("`%s` is not defined, perhaps `import %s;` is needed?", exp.ident.toChars(), n.ptr);
```
Comment (jacob-carlborg) [inline src/dmd/expressionsem.d]: Isn't it better to use `%.*` and pass the length and pointer separately?

Comment (ibuclaw) [reply src/dmd/expressionsem.d]: Was taking advantage of that strings from importHint are null terminated.  Can do though.

**Comment by thewilsonator** [comment]
The autotester OOM'd on FreeBSD.

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (jacob-carlborg) [reply src/dmd/expressionsem.d]: I’m guessing it won’t be null terminated after most code uses D strings.

**Review by jacob-carlborg — APPROVED** [review]

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**